src/openMVG/sfm/sfm_data.cpp

```diff
@@ -117,7 +117,10 @@ bool ColorizeTracks( SfM_Data & sfm_data )
       if (it != obs.end())
       {
         // Color the track
-        const Vec2 & pt = it->second.x;
+        Vec2 pt = it->second.x;
+        // Clamp the pixel position if the feature/marker center is outside the image.
+        pt.x() = clamp(pt.x(), 0.0, double(image.Width()-1));
```

Comment:
Ok, I better understand, and you do not want to reject those detection too early.